tools/clang/test/HLSLFileCheck/hlsl/functions/arguments/maybe_unused_out.hlsl

```diff
@@ -0,0 +1,10 @@
+// RUN: %dxc -T lib_6_4 %s -ast-dump | FileCheck %s
+// Verify attribute annotation to opt out of uninitialized parameter analysis.
+
+void UnusedOutput([maybe_unused] out int Val) {}
```

Comment:
Do we need to test this for every possible way this error can be produced? Maybe a variant of the above that ensures this sidesteps any such warnings?

I think we need a test for the disabling flag

Reply:
Adding more robust testing around the attribute makes sense.

Generally clang doesn't add per-warning or per-warning group tests for disabling warnings because the general infrastructure is tested, and that adds a lot of potential tests with minimal added value.

Since DXC likely isn't running the warning disabling tests, it probably also makes sense to add tests for that case too.